Adding section to migrate and use OCI layout format (#612)
buildpack/docs: content/docs/reference/spec/migration/platform-api-0.11-0.12.md

TODO

### OCI layout is a supported export format (experimental)

> In Platform 0.12 a new capability to [export an application images on disk in OCI format](https://github.com/buildpacks/rfcs/blob/main/text/0119-export-to-oci.md) was added.

Reviewer: ```suggestion
In Platform 0.12, a new capability to [export application images on disk in OCI layout format](https://github.com/buildpacks/rfcs/blob/main/text/0119-export-to-oci.md) was added.
```

> Platform must take care of preparing a [local storage](https://github.com/buildpacks/rfcs/blob/main/text/0119-export-to-oci.md#how-it-works) on disk where images are being saved in OCI layout format and passing it through to the lifecycle.

Reviewer: ```suggestion
Platform must prepare a [layout directory](https://github.com/buildpacks/rfcs/blob/main/text/0119-export-to-oci.md#how-it-works) containing input images in OCI layout format, and provide the location of the directory to the lifecycle.
```

#### Lifecycle phases affected

The lifecycle phases affected by this new behavior are:
- [Analyze](https://buildpacks.io/docs/concepts/components/lifecycle/analyze/)
- [Restore](https://buildpacks.io/docs/concepts/components/lifecycle/restore/)
- [Export](https://buildpacks.io/docs/concepts/components/lifecycle/export/)
- [Create](https://buildpacks.io/docs/concepts/components/lifecycle/create/)

#### Before executing the lifecycle

> Input images required by any phase, like the `run-image`, must be saved on disk in OCI layout format in the local storage previously defined following the 

Reviewer: Is it just run image and previous image? We don't allow cache image in OCI layout do we?

[rules](https://github.com/buildpacks/spec/blob/platform/v0.12/platform.md#map-an-image-reference-to-a-path-in-the-layout-directory) to convert the reference to a path.

#### During lifecycle execution

> For the phases affected, the feature is enabled using a new flag `-layout` or setting to **true** an environment variable named `CNB_USE_LAYOUT`.

Reviewer: ```suggestion
For the phases affected, the feature is enabled by providing a new `-layout` flag or by setting the `CNB_USE_LAYOUT` environment variable to `true`.
```

* If the feature is enabled: 

>   *  A path to a directory where the images are saved in OCI layout format must be specified, either by setting the flag `-layout-dir` or the environment variable `CNB_LAYOUT_DIR`.

Reviewer: ```suggestion
  *  A path to a directory where the images are located and saved must be specified, either by providing a `-layout-dir` flag or by setting the `CNB_LAYOUT_DIR` environment variable.
```

Reviewer: Should we also mention that `rebase` isn't supported, and you can't use this feature with image extensions?

Author: Yeah, I think that is important to mention

- [Export](https://buildpacks.io/docs/concepts/components/lifecycle/export/)
- [Create](https://buildpacks.io/docs/concepts/components/lifecycle/create/)

> **Note** [Rebase](https://buildpacks.io/docs/concepts/components/lifecycle/rebase/) an image exported to OCI layout format is not supported.

Reviewer: ```suggestion
**Note** [Rebasing](https://buildpacks.io/docs/concepts/components/lifecycle/rebase/) an image exported to OCI layout format is not supported.
```

#### Before executing the lifecycle

> Input images required by any phase, like the `run-image` or `previous-image`, must be saved on disk in OCI layout format in the local storage previously defined following the 

Reviewer: ```suggestion
Input images required by any phase, like the `run-image` or `previous-image`, must be saved on disk in OCI layout format in the layout directory following the 
```
